clang/include/clang/Basic/Attr.td

```diff
@@ -3039,9 +3039,12 @@ def WorkGroupSizeHint :  InheritableAttr {
   let Spellings = [GNU<"work_group_size_hint">,
                    CXX11<"sycl", "work_group_size_hint">];
   let Args = [ExprArgument<"XDim">,
-              ExprArgument<"YDim">,
-              ExprArgument<"ZDim">];
+              ExprArgument<"YDim", /*optional*/1>,
+              ExprArgument<"ZDim", /*optional*/1>];
   let Subjects = SubjectList<[Function], ErrorDiag>;
+  // In OpenCL C this attribute takes no default values wherehas in SYCL it does. 
+  // To avoid confusing diagnostics, the checks are deffered to "handleWorkGroupSizeHint". 
```

Comment:
```suggestion
  // In OpenCL C this attribute takes no default values whereas in SYCL it does. 
  // To avoid confusing diagnostics, the checks are deferred to "handleWorkGroupSizeHint". 
```